opentelemetry-api/src/opentelemetry/metrics/_internal/instrument.py

```diff
@@ -180,7 +195,7 @@ class Counter(Synchronous):
     def add(
         self,
         amount: Union[int, float],
-        attributes: Optional[Attributes] = None,
+        attributes: Optional[_MetricAttributes] = None,
```

Comment:
TODO: we can use `Attributes` and let users use complex attributes on metrics without any restrictions. We can also introduce an alias for a limited attribute set.

Reply:
I like the idea of allowing it at runtime but blocking it at type checking time. It gives users feedback that they're doing something probably wrong but is easy to override.